### DISCUSSION TITLE
Re-directed to google account selection screen in a loop

### DISCUSSION THREAD

**Discussion by abudhakeer** [supabase/supabase#7657] [Questions] [answered]
I see this weird issue sometimes in chrome but always in firefox that even after i selected a particular account in the google account selection screen am still getting redirected to the same page in a loop. On further debugging i found that this loop will happen until i receive the `onAuthStateChanged` event. How can i avoid this state?

https://user-images.githubusercontent.com/2256033/178440228-e3da0ad0-6f2c-4d9c-bec2-b4c9753612e7.mov

Attaching the video for reference.

**Comment by hf** [comment, marked as answer]
Hi, would you mind sharing a recording of the network requests that occur on this page. Additionally, make sure that the page that Google redirects back to does not immediately trigger a `signIn` API call. It's often best practice to have a button that the user clicks to trigger a `signIn` action.

**Reply by abudhakeer** [reply]
@hf Thank you for pointing out that the `signIn`call should be an invocation of a button click. That was the issue. Fixed it now :)
